gammapy/maps/hpxnd.py

```diff
@@ -452,6 +457,130 @@ def smooth(self, width, kernel="gauss"):
 
         return self._init_copy(data=smoothed_data)
 
+    def convolve_wcs(self, kernel, use_fft=True, **kwargs):
+        """
+        Convolve map with a WCS kernel.
+
+        It projects the map into a WCS geometry, convolves with a WCS kernel and
+        projects back into the initial Healpix geometry.
+
+        If the kernel is two dimensional, it is applied to all image planes likewise.
+        If the kernel is higher dimensional it must match the map in the number of
+        dimensions and the corresponding kernel is selected for every image plane.
+
+        Parameters
+        ----------
+        kernel : `~gammapy.irf.PSFKernel`
+            Convolution kernel. The pixel size must be upsampled by a factor 2 or bigger
+            with respect to the input map to prevent artifacts in the projection.
+        use_fft : bool
+            Use `scipy.signal.fftconvolve` if True (default)
+            and `ndi.convolve` otherwise.
+        kwargs : dict
+            Keyword arguments passed to `scipy.signal.fftconvolve` or
+            `ndi.convolve`.
+
+        Returns
+        -------
+        map : `HpxNDMap`
+            Convolved map.
+        """
+        if self.geom.width > 10*u.deg:
+            log.warning("Convolution via WCS projection is not recommended for large maps. "\
+                "Perhaps the method `convolve_full()` is more suited for this case.")
+        wcs_geom = kernel.psf_kernel_map.geom.to_image()
+        wcs_size = np.min(wcs_geom._cdelt)
+        hpx_size = get_pix_size_from_nside(self.geom.nside[0])
+        if wcs_size > 0.5*hpx_size:
+            raise ValueError("The kernel pixel size has to be smaller by at least a factor 2 than the input map to prevent artifacts.")
+        hpx2wcs = HpxToWcsMapping.create(self.geom, wcs_geom)
+        # Project to WCS and convolve
+        wcs_map  = self.to_wcs(hpx2wcs = hpx2wcs, fill_nan = False)
+        conv_wcs_map = wcs_map.convolve(kernel, use_fft,**kwargs)
+        # and back to hpx
+        hpx_data = HpxNDMap.from_geom(self.geom).data
+        hpx_data = hpx2wcs.fill_hpx_map_from_wcs_data(conv_wcs_map.data, hpx_data)
+        conv_hpx_map = HpxNDMap.from_geom(self.geom, data = hpx_data)
+        return conv_hpx_map
+
+    def convolve_full(self, kernel):
+        """
+        Convolve map with a symmetrical WCS kernel.
+
+        It extracts the radial profille of the kernel (assuming radial symmetry) and
+        convolves via `hp.sphtfunc.smoothing`. Since no projection is applied, this is
+        suited for full-sky and large maps.
+
+        If the kernel is two dimensional, it is applied to all image planes likewise.
+        If the kernel is higher dimensional it must match the map in the number of
+        dimensions and the corresponding kernel is selected for every image plane.
+
+        Parameters
+        ----------
+        kernel : `~gammapy.irf.PSFKernel`
```

Comment:
In general the radial profile is stored directly in a `PSFMap`, so we could consider supporting a `PSFMap` instead of `PSFKernel`...but only if we decide the `.convolve_full()` is useful to keep.

Reply:
Yes, I was a bit annoyed to be getting the radial profile when the kernel was built from a radial profile... it is obviously more efficient to skip the kernel computation completely. But I wasn't sure if that would work/be worth it for this case, so kept the kernel for the time being.